src/utils.c

```diff
@@ -384,3 +390,62 @@ date_bucket(PG_FUNCTION_ARGS)
 	bucketed = DirectFunctionCall2(timestamp_bucket, PG_GETARG_DATUM(0), converted_ts);
 	return DirectFunctionCall1(timestamp_date, bucketed);
 }
+
+/* Returns approximate period in microseconds */
+int64
+get_interval_period_approx(Interval *interval)
+{
+	return interval->time + (((interval->month * 30) + interval->day) * USECS_PER_DAY);
+}
+
+/* Returns approximate period in microseconds */
+int64
+date_trunc_interval_period_approx(text *units)
+{
+	int			decode_type,
+				val;
+	char	   *lowunits = downcase_truncate_identifier(VARDATA_ANY(units),
+														VARSIZE_ANY_EXHDR(units),
+														false);
+
+	decode_type = DecodeUnits(0, lowunits, &val);
+
+	if (decode_type != UNITS)
+		return -1;
+
+	switch (val)
+	{
+		case DTK_WEEK:
+			return 7 * USECS_PER_DAY;
+		case DTK_MILLENNIUM:
+			return 1000 * 356 * USECS_PER_DAY;
```

Comment:
Same here, prefer `#define`s like `DAYS_PER_YEAR` if such defines do not already exist.

Reply:
Fixed